src/config.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Handles a configuration dictionary.
+ *
+ * @class Config
+ * @extends Model
+ */
+
+CKEDITOR.define( [ 'ckeditor', 'mvc/model', 'utils' ], function( CKEDITOR, Model, utils ) {
+	var Config = Model.extend( {
+		/**
+		 * Creates an instance of the {@link Config} class.
+		 *
+		 * @param {Object} [configurations] The initial configurations to be set.
+		 * @constructor
+		 */
+		constructor: function Config( configurations ) {
+			// Call super-constructor.
+			Model.apply( this );
+
+			if ( configurations ) {
+				this.set( configurations );
+			}
+		},
+
+		/**
+		 * Set configuration values.
+		 *
+		 * It accepts both a name/value pair or an object, which properties and values will be used to set
+		 * configurations.
+		 *
+		 * It also accepts setting a "deep configuration" by using dots in the name. For example, `'resize.width'` sets
+		 * the value for the `width` configuration in the `resize` subset.
+		 *
+		 *     config.set( 'width', 500 );
+		 *     config.set( 'toolbar.collapsed', true );
+		 *
+		 *     // Equivalent to:
+		 *     config.set( {
+		 *         width: 500
+		 *         toolbar: {
+		 *             collapsed: true
+		 *         }
+		 *     } );
+		 *
+		 * Passing an object as the value will amend the configuration, not replace it.
+		 *
+		 *     config.set( 'toolbar', {
+		 *         collapsed: true,
+		 *     } );
+		 *
+		 *     config.set( 'toolbar', {
+		 *         color: 'red',
+		 *     } );
+		 *
+		 *     config.toolbar.collapsed; // true
+		 *     config.toolbar.color; // 'red'
+		 *
+		 * @param {String|Object} nameOrConfigurations The configuration name or an object from which take properties as
+		 * configuration entries.
+		 * @param {*} [value] The configuration value. Used if a name is passed to nameOrConfigurations. If undefined,
```

Comment:
> What about def value [value=null] and shorter descr?

Yup, this is the right syntax, we used it before already and it should be used here as well.